### PR TITLE
add brief overview of Maven vs Gradle, add links

### PR DESCRIPTION
**Pull request by linda-seiter**
Moved lesson to module 4.  Minor updates to introduction.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,8 +30,27 @@ for:
 6. Cross Platform: Allows projects to be easily managed across different
    environments.
 
+## Maven and Gradle
+
+Maven and Gradle are two popular build automation tools. 
+
+- Maven is based on developing Java projects. Gradle is based on developing
+domain-specific language (DSL) projects and can build applications of any kind, including Java.
+- Maven uses XML to structure a project. Gradle uses a DSL based on Groovy or Kotlin.
```
Comment (kcleland0818) [inline README.md]: Should we explain more on what Groovy and Kotlin are? Or do you plan to elaborate on them a little when you cover Gradle?

**Review by kcleland0818 — COMMENTED** [review]
Just a question! Looks good!

File: README.md (hunk 1 above)
Comment (linda-seiter) [reply README.md]: I just added a brief explanation to this lesson and will provide more explanation in the Gradle lesson.

**Review by kcleland0818 — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (kcleland0818) [reply README.md]: Thanks!

**Review by AlveeM — APPROVED** [review]

**Pull request merged**